bin/setupReplication.js

```diff
@@ -0,0 +1,431 @@
+            return cb(null, res);
+        });
+    }
+
+    _enableVersioning(where, cb) {
+        const bucket = where === 'source' ? this._sourceBucket :
+            this._targetBucket;
+        const params = {
+            Bucket: bucket,
+            VersioningConfiguration: {
+                Status: 'Enabled',
+            },
+        };
+        this._s3Clients[where].putBucketVersioning(params, (err, res) => {
+            if (err) {
+                this._log.error('error enabling versioning', {
+                    error: err,
+                    method: '_SetupReplication._enableVersioning',
+                    errorMessage: err.message,
+                    errorStack: err.stack,
+                });
+                return cb(err);
+            }
+            this._log.debug('Versioning enabled', {
+                bucket: where,
+                response: res,
+                method: '_enableVersioning',
+            });
+            return cb(null, res);
+        });
+    }
+
+    _attachResourcePolicy(policyArn, roleName, where, cb) {
+        const params = {
+            PolicyArn: policyArn,
+            RoleName: roleName,
+        };
+        this._iamClients[where].attachRolePolicy(params, (err, res) => {
+            if (err) {
+                this._log.error('error attaching resource policy', {
+                    error: err,
+                    method: '_SetupReplication._attachResourcePolicy',
+                    errorMessage: err.message,
+                    errorStack: err.stack,
+                });
+                return cb(err);
+            }
+            this._log.debug('Attached resource policy', {
+                bucket: where,
+                response: res,
+                method: '_attachResourcePolicy',
+            });
+            return cb(null, res);
+        });
+    }
+
+    _enableReplication(roleArns, cb) {
+        const params = {
+            Bucket: this._sourceBucket,
+            ReplicationConfiguration: {
+                Role: roleArns,
+                Rules: [{
+                    Destination: {
+                        Bucket: `arn:aws:s3:::${this._targetBucket}`,
+                    },
+                    Prefix: '',
+                    Status: 'Enabled',
+                }],
+            },
+        };
+        this._s3Clients.source.putBucketReplication(params, (err, res) => {
+            if (err) {
+                this._log.error('error enabling replication', {
+                    error: err,
+                    method: '_SetupReplication._enableReplication',
+                    errorMessage: err.message,
+                    errorStack: err.stack,
+                });
+                return cb(err);
+            }
+            this._log.debug('Bucket replication enabled', {
+                response: res,
+                method: '_enableReplication',
+            });
+            return cb(null, res);
+        });
+    }
+
+    _parallelTasks(cb) {
```

Comment:
This function may be better named with a more descriptive name like `_createEntities` which tells more about what it does (external info) than how it does it (internal info).

Alternatively it could also be clearer to put the function body in the main `run()` method, since all steps appearing in a single place are easier to follow (and since all of them are one-liner calls to sub methods it should not grow the `run` method beyond a reasonable size)